src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -98,6 +100,9 @@ public VisualStudioWorkspaceImpl(ExportProvider exportProvider, IAsyncServicePro
             _projectionBufferFactoryService = exportProvider.GetExportedValue<IProjectionBufferFactoryService>();
             _projectCodeModelFactory = exportProvider.GetExport<IProjectCodeModelFactory>();
 
+            var vsServiceProvider = exportProvider.GetExport<SVsServiceProvider>().Value;
+            _vsSolution = vsServiceProvider.GetService<IVsSolution>();
```

Comment:
Unless this has been confirmed to be a free-threaded service this can't be written in a MEF constructor since this may construct on a background thread and then this would deadlock.

Reply:
Is this ok?
```C#
_lazyVsSolution = new Lazy<IVsSolution>(() => exportProvider.GetExport<SVsServiceProvider>().Value.GetService<IVsSolution>());
```

Reply:
That should have been SVsSolution, not IVsSolution.

Reply:
OK, I'll fix it up.